src/connectors/answers/connectAnswers.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  checkRendering,
+  createDocumentationMessageGenerator,
+  createConcurrentSafePromise,
+  addQueryID,
+  debounce,
+  addAbsolutePosition,
+  noop,
+  escapeHits,
+} from '../../lib/utils';
+import {
+  Connector,
+  Hits,
+  Hit,
+  FindAnswersOptions,
+  FindAnswersResponse,
+} from '../../types';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'answers',
+  connector: true,
+});
+
+export type AnswersRendererOptions = {
+  /**
+   * The matched hits from Algolia API.
+   */
+  hits: Hits;
+
+  /**
+   * whether it's still loading the results from the /answers API
+   */
+  isLoading: boolean;
+};
+
+export type AnswersConnectorParams = {
+  /**
+   * Attributes to use for predictions.
+   * If empty, we use all `searchableAttributes` to find answers.
+   * All your `attributesForPrediction` must be part of your `searchableAttributes`.
+   */
+  attributesForPrediction?: string[];
+
+  /**
+   * The languages in the query. Currently only supports `en`.
+   */
+  queryLanguages: Array<'en'>;
+
+  /**
+   * Maximum number of answers to retrieve from the Answers Engine.
+   * Cannot be greater than 1000.
+   */
+  nbHits?: number;
+
+  /**
+   * Debounce time in milliseconds to debounce render
+   * @default 100
+   */
+  renderDebounceTime?: number;
+
+  /**
+   * Debounce time in milliseconds to debounce search
+   * @default 100
+   */
+  searchDebounceTime?: number;
+
+  /**
+   * Whether to escape HTML tags from hits string values.
+   *
+   * @default true
+   */
+  escapeHTML?: boolean;
+
+  /**
+   * Extra parameters to pass to findAnswers method.
+   * @default {}
+   */
+  extraParameters?: FindAnswersOptions;
+};
+
+export type AnswersConnector = Connector<
+  AnswersRendererOptions,
+  AnswersConnectorParams
+>;
+
+const connectAnswers: AnswersConnector = function connectAnswers(
+  renderFn,
+  unmountFn = noop
+) {
+  checkRendering(renderFn, withUsage());
+
+  return widgetParams => {
+    const {
+      queryLanguages,
+      attributesForPrediction,
+      nbHits = 1,
+      renderDebounceTime = 100,
+      searchDebounceTime = 100,
+      escapeHTML = true,
+      extraParameters = {},
+    } = widgetParams || ({} as typeof widgetParams);
+
+    if (!queryLanguages || queryLanguages.length === 0) {
+      throw new Error(
+        withUsage('The `queryLanguages` expects an array of strings.')
+      );
+    }
+
+    const runConcurrentSafePromise = createConcurrentSafePromise<
+      FindAnswersResponse<Hit>
+    >();
+
+    let lastResult: Partial<FindAnswersResponse<Hit>>;
+    let isLoading = false;
+    const debouncedRender = debounce(renderFn, renderDebounceTime);
+    let debouncedRefine;
+
+    return {
+      $$type: 'ais.answers',
+
+      init(initOptions) {
+        const { state, instantSearchInstance } = initOptions;
+        const answersIndex = instantSearchInstance.client!.initIndex!(
```

Comment:
Just checking but is it a use case to use another search client (i.e. another application  ID) for Answers?

Reply:
at the moment there would be use cases for that as well in multi-index, but we decided that you should use a custom client to merge the requests. I think that for now at least we should do the same with answers

Reply:
Then we should do the same with React InstantSearch, which [does accept a `searchClient` prop](https://github.com/algolia/react-instantsearch/blob/0228c51c2d65e477fbce6471e04a85597f5bfda6/packages/react-instantsearch-dom/src/hooks/useAnswers.js#L13) right now.

Reply:
> Just checking but is it a use case to use another search client (i.e. another application ID) for Answers?

I initially misunderstood your question. I'm not so sure about another application ID. We do not have any case using different application IDs under a single instantSearchInstance. For multi-index, we should support <index> widget, though.

The reason why we're accepting `searchClient` in React InstantSearch is because I don't have access to the `client` like I do in IS.js. Not sure if we need to require another prop on the vanilla JS to make it look the same as React IS.

By the way, the next step for Answers widget is to pass all the facets from the current widget tree down to Answers API call, so I think it makes more sense to keep things under the same application ID.

Reply:
> The reason why we're accepting `searchClient` in React InstantSearch is because I don't have access to the `client` like I do in IS.js. Not sure if we need to require another prop on the vanilla JS to make it look the same as React IS.

I don't think that this current limitation should have such an impact on our API then. Getting the search client in the React `Answers` widget should be possible.

> By the way, the next step for Answers widget is to pass all the facets from the current widget tree down to Answers API call, so I think it makes more sense to keep things under the same application ID.

That makes sense!

Reply:
> Getting the search client in the React Answers widget should be possible.
Not sure if I understood you correctly.
Do you mean it should be possible without explicitly accepting the search client?

Reply:
I mean that we should be able to get the search client via the React Context with some modifications. A technical limitation that we can solve shouldn't impact the user API. But we can discuss this on the React InstantSearch PR.

Reply:
@francoischalifour I agree. Let's continue on the React IS PR.